File: 0966-binary-subarrays-with-sum/0966-binary-subarrays-with-sum.java
class Solution {
    public int numSubarraysWithSum(int[] nums, int goal) {
         int res =0;
        for (int i =0 ;i<nums.length; i++){
            int sum = nums[i];
            if (sum == goal) res++;
            for (int j = i+1;j<nums.length;j++){
                 sum+=nums[j];
                if (sum == goal) res++; 
            }
        }
        return res;
    }
}

File: 0966-binary-subarrays-with-sum/0966-binary-subarrays-with-sum.cpp
class Solution {
public:
int numSubarraysWithSum(vector<int>& nums, int goal) {        unordered_map <int, int> count;
//hash map
        count[0]=1;
         int presum=0;
         int totalsubarr=0;
          for(auto i:nums){
        presum+=i;
        if(count.find(presum-goal)!=count.end())
        {   totalsubarr+=count[presum-goal];}
        count[presum]++;
          }
    return totalsubarr;
}
};
